analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/InitializationVectorShouldBeRandomBase.cs

```diff
@@ -70,7 +70,7 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
         && property.Instance is { } propertyInstance
         && state.ResolveCaptureAndUnwrapConversion(propertyInstance).TrackedSymbol() is { } propertyInstanceSymbol
         && IsIVProperty(property, propertyInstanceSymbol)
-        && state[assignment.Value]?.HasConstraint(ByteCollectionConstraint.CryptographicallyWeak) is { }
+        && state[assignment.Value]?.HasConstraint(ByteCollectionConstraint.CryptographicallyWeak) is true
```

Comment:
This change is not tested. We have to add an UT.
`is { }` allows both, `is true` and `is false`, therefore `is false` scenario is not tested. 

To be extra clear: If I change it back to `is { }`, the new UT should fail.